m3u8/structure.go

```diff
@@ -117,34 +117,41 @@ const (
-	independentSegments bool            // Global tag for EXT-X-INDEPENDENT-SEGMENTS
-
+	TargetDuration      uint              // TargetDuration is max media segment duration. Rounding depends on version.
+	SeqNo               uint64            // EXT-X-MEDIA-SEQUENCE
+	Segments            []*MediaSegment   // List of segments in the playlist. Output may be limited by winsize.
+	Args                string            // optional query placed after URIs (URI?Args)
+	Defines             []Define          // EXT-X-DEFINE tags
+	Iframe              bool              // EXT-X-I-FRAMES-ONLY
+	Closed              bool              // is this VOD/EVENT (closed) or Live (sliding) playlist?
+	MediaType           MediaType         // EXT-X-PLAYLIST-TYPE (EVENT, VOD or empty)
+	DiscontinuitySeq    uint64            // EXT-X-DISCONTINUITY-SEQUENCE
+	StartTime           float64           // EXT-X-START:TIME-OFFSET=<n> (positive or negative)
+	StartTimePrecise    bool              // EXT-X-START:PRECISE=YES
+	Key                 *Key              // EXT-X-KEY is initial key tag for encrypted segments
+	Map                 *Map              // EXT-X-MAP provides a Media Initialization Section. Segments can redefine.
+	DateRanges          []*DateRange      // EXT-X-DATERANGE tags not associated with SCTE-35
+	AllowCache          *bool             // EXT-X-ALLOW-CACHE tag YES/NO, removed in version 7
+	Custom              CustomMap         // Custom-provided tags for encoding
+	customDecoders      []CustomDecoder   // customDecoders provides custom tags for decoding
+	winsize             uint              // max number of segments encoded sliding playlist, set to 0 for VOD and EVENT
+	capacity            uint              // total capacity of slice used for the playlist
+	head                uint              // head of FIFO, we add segments to head
+	tail                uint              // tail of FIFO, we remove segments from tail
+	count               uint              // number of segments added to the playlist
+	buf                 bytes.Buffer      // buffer used for encoding and caching playlist output
+	scte35Syntax        SCTE35Syntax      // SCTE-35 syntax used in the playlist
+	ver                 uint8             // protocol version of the playlist, 3 or higher
+	targetDurLocked     bool              // target duration is locked and cannot be changed
+	independentSegments bool              // Global tag for EXT-X-INDEPENDENT-SEGMENTS
+	PartTargetDuration  float64           // EXT-X-PART-INF:PART-TARGET
+	PartialSegments     []*PartialSegment // List of partial segments in the playlist.
+	NextMSNIndex        uint64            // The index to be used for the next full segment
+	NextPartIndex       uint64            // The index to be used for the next partial segment (starts at 0)
+	MaxPartIndex        uint64            // To determine when to "roll over" on the NextPartIndex
```

Comment:
I think the documentation should be better on the `MaxPartIndex` field.

Something like `The maximum index of a partial segment` rather than the usage.

Then, the whole usage of the field should probably be described somewhere, maybe in a section in
the README file how low-latency works when reading and writing. It should also describe how SKIP works in read and write mode.